src/profile-logic/process-profile.js

```diff
@@ -405,15 +522,13 @@ function addProcessedTaskTracerData(tasktracer, result, libs, startTime) {
  * subprocess profiles into the parent process profile; each profile's process
  * has its own timebase, and we don't want to keep converting timestamps when
  * we deal with the integrated profile.
- * @param {object} samples The table of samples/markers.
- * @param {number} delta The time delta, in milliseconds, by which to adjust.
- * @return {object} A samples table with adjusted time values.
  */
-function adjustSampleTimestamps(samples, delta) {
+function _adjustSampleTimestamps(
+  samples: SamplesTable,
+  delta: Milliseconds
+): SamplesTable {
   return Object.assign({}, samples, {
-    time: samples.time.map(
-      time => (time === undefined ? undefined : time + delta)
-    ),
+    time: samples.time.map(time => time + delta),
```

Comment:
At the moment, all samples have times, as far as I can tell. It's possible that old profiles did not have time values for some samples; if we find any of them, we can add something to the profile version upgrading code. But for now let's remove the check here.